src/ocs/CommandListener.cpp

```diff
@@ -618,11 +620,9 @@ string CommandListener::get_device(string name) {
 
 int main() { 
     CommandListener cmd; 
-    usleep(10000000); 
-    /**
     while (1) { 
-
+    
+        this_thread::sleep_for(chrono::milliseconds(1));
```

Comment:
I understand this is a temporary fix, since this really should be joining the threads that are started, but why not just use the system call "sleep()" instead?

Reply:
At the time, the thinking was I want to sleep the application only for some mili-seconds. system call `sleep` only supports times in seconds. Should I be worried about having a latency of 1 second between every message or is just sleeping for 1 second okay?

Reply:
The threads run independently of the main thread.  You can sleep for a very long time, and none of the other threads will be impacted.  In fact, you are impacting those threads more by waking up and looping more frequently. You should be able to change this to 60 seconds, and your program should run exactly the same.  That's for this part of the code.  The other "sleep" type of operations in the code are intended to do something else, but that's a different issue.  In some of those cases, we're waiting for setup of communication channels through callbacks, and there are alternate ways of dealing with those issues.   That kind of a issue, and others, need to be looked at in a case by case basis, later, in another ticket.

Reply:
That clears my confusion. Thanks Steve. I will fix the code as suggested.